python/pyspark/sql/types.py

```diff
@@ -438,6 +438,11 @@ def toInternal(self, obj):
     def fromInternal(self, obj):
         return self.dataType.fromInternal(obj)
 
+    def typeName(self):
+        raise TypeError(
+            "StructField does not have typename. \
+            You can use self.dataType.simpleString() instead.")
```

Comment:
I'd remove `self` here and just say something like ` use typeName() on its type explicitly ...`.

Reply:
use mean `simpleString()` and not `typeName()`. right ?

Reply:
I think it should be `typeName` on its datatype because `typeName` was called.